src/Controls/src/Core/Toolbar/Toolbar.iOS.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using CoreGraphics;
+using Microsoft.Maui.Graphics;
+using UIKit;
+using PointF = CoreGraphics.CGPoint;
+using RectangleF = CoreGraphics.CGRect;
+using SizeF = CoreGraphics.CGSize;
+
+namespace Microsoft.Maui.Controls
+{
+	public partial class Toolbar
+	{
+		NavigationManager? NavigationManager => Handler.MauiContext?.GetNavigationManager();
+
+		UINavigationController? NavigationController => NavigationManager?.NavigationController;
+
+		public static void MapIsVisible(IToolbarHandler handler, Toolbar toolbar)
+		{
+			if (toolbar == null)
+			{
+				return;
+			}
+
+			if (toolbar?.NavigationController == null)
+			{
+				throw new NullReferenceException("NavigationController is null.");
+			}
+
+			if (toolbar.NavigationController.NavigationBarHidden == toolbar.IsVisible)
+			{
+				// TODO: the view underneath this jumps up when the toolbar is hidden and down when it's shown
+				// rather than smoothly animating the transition. It's not clear how to fix this
+				// Also, when the toolbar is hidden and another page is pushed, the toolbar for the page that's about to hide
+				// gets its toolbar back right before the animation of the push of the new page
+				toolbar.NavigationController.SetNavigationBarHidden(!toolbar.IsVisible, true);
```

Comment:
Added some notes here - this doesn't work smoothly for the reasons I spell out in the comments. I've tried some things to try and replicate the Renderer's approach but no luck - maybe this is one of the things that needs the ParentingViewController approach to work?